app/controllers/oaisys/pmh_controller.rb

```diff
@@ -2,25 +2,84 @@
 
 module Oaisys
   class PMHController < ApplicationController
+    EXPECTED_ARGUMENTS = {
+        identify: {
+            required: %w(),
+            optional: %w(),
+            exclusive: %w()
+        },
+        list_sets: {
+            required: %w(),
+            optional: %w(),
+            exclusive: %w(resumptionToken)
+        },
+        list_metadata_formats: {
+            required: %w(),
+            optional: %w(identifier),
+            exclusive: %w()
+        },
+        list_records: {
+            required: %w(metadataPrefix),
+            optional: %w(from until set),
+            exclusive: %w(resumptionToken)
+        },
+        get_record: {
+            required: %w(identifier metadataPrefix),
+            optional: %w(),
+            exclusive: %w()
+        },
+        list_identifiers: {
+            required: %w(metadataPrefix),
+            optional: %w(from until set),
+            exclusive: %w(resumptionToken)
+        },
+    }.freeze
     def bad_verb
     end
 
     def identify
+      expected_ags(__method__)
     end
 
     def list_sets
+      expected_ags(__method__)
     end
 
     def list_metadata_formats
+      expected_ags(__method__)
     end
 
     def list_records
+      expected_ags(__method__)
     end
 
     def get_record
+      expected_ags(__method__)
     end
 
     def list_identifiers
+      expected_ags(__method__)
+    end
+
+    private
+
+    def expected_ags (verb)
+      @verb = verb.to_s.camelize
+      @error_code = 'badArgument'
+      @error_message = 'The request includes illegal arguments or is missing required arguments.'
```

Comment:
I could really go either way on this, but thoughts on putting this kind of message into the localization file instead?

I know we're not exactly likely to localize the API, but it's a nice way to cleanly separate out constant strings. Thoughts?

Reply:
I'm up for having/using a localization file. How do we determine what goes in it and what doesn't though? Couldn't the error code 'badArgument' go in it as well?

Reply:
Yeah, I'm not going to lie, it's a lot of gut feel and good taste, which isn't a satisfying explanation.

The message, to me, feels more "human readable" -- it has meaning to humans, we might want to change it, it's a long string in the code (which is always kind of a code smell) ... it _feels_ like something that shouldn't be hardcoded, so it could change pretty independently of the codebase.

The error code is more "intended for machines" -- it has a meaning in the scope of the standard, it's essentially a constant of sorts in that short of a change in the standard it should never change, and it has a unique identity rather than an instanced one -- it's probably the kind of thing that's better represented by a symbol than a string, actually, and in other languages we'd tend to leverage something like an enum to represent it. 

So to me, it _doesn't_ make sense to pull it out into a separate place that's intended for allowing human-meaningful strings to easily change without touching code. Does that make sense?

Reply:
Yes, makes sense! I've made the change.